### PR TITLE
BUG: Handle base case for scipy.integrate.simpson when span along the axis is 0 

### LINKED ISSUE #15721: BUG: scipy.integrate.simps returns NaN when span of x is 0
### Describe your issue.

Obviously, this is an edge case, but I think it should be handled. Having an array for x with no span (all values are the same), integrate.simpson returns NaN and throws a couple divide by 0 errors from _quadrature. Instead, since integrating over a span of 0 is in fact 0, I think the function should return 0.

### Reproducing Code Example

```python
scipy.integrate.simpson(range(5), [0,0,0,0,0])
```

### Error message

```shell
C:\AeroApps\AnacondaP3x64-5.1.0\envs\cf310\lib\site-packages\scipy\integrate\_quadrature.py:415: RuntimeWarning: invalid value encountered in true_divide
  h0divh1 = h0 / h1
C:\AeroApps\AnacondaP3x64-5.1.0\envs\cf310\lib\site-packages\scipy\integrate\_quadrature.py:417: RuntimeWarning: invalid value encountered in true_divide
  y[slice1] * (hsum * hsum / hprod) +
Out[58]: nan
```

### SciPy/NumPy/Python version information

1.8.0 1.22.3 sys.version_info(major=3, minor=10, micro=2, releaselevel='final', serial=0)

### PR DESCRIPTION
**Pull request by k-styles**
#### Reference issue

Closes #15721 

#### What does this implement/fix?

As described in the issue, the current implementation of `scipy.integrate.simpson` return `NaN` for when the span along the integrable axis is `0`. 
I change the code so that `0` is returned for that situation instead of `NaN`.

### CODE REVIEW

**Comment by k-styles** [comment]
@tupui Let me know, if except the checks, the code will be working fine.

**Review by tupui — CHANGES_REQUESTED** [review]
Thank you @k-styles for the PR. A lot of tests seem to be broken with the changes, it's not really my area so without digging into it there is not much I can say for now. A test is also missing to check the new behaviour.

**Comment by k-styles** [comment]
@tupui All workflows seems to have passed.

**Comment by k-styles** [comment]
@tupui Are there any more changes I need to make?

**Review by tupui — CHANGES_REQUESTED** [review]
@k-styles, as I said before you are missing a test showing that the change is doing what you want. It has to work with different axis, and all other parameters.

**Comment by k-styles** [comment]
@tupui Oh okay, I wasn't aware I need to add that as well.
I'll do it shortly.

**Comment by Patol75** [comment]
Wouldn't it be simpler to directly test `x` when `simpson` is called? Probably something along the lines of
```
if x is not None and np.count_nonzero(np.diff(x, axis=axis)) == 0:
    return 0
```
should work.

**Comment by tupui** [comment]
> Wouldn't it be simpler to directly test `x` when `simpson` is called? Probably something along the lines of
> 
> ```
> if x is not None and np.count_nonzero(np.diff(x, axis=axis)) == 0:
>     return 0
> ```
> 
> should work.

Ideally, the solution should work for any constant, not just 0. This was discussed in the previous PR or issue AFAIR

**Comment by Patol75** [comment]
`count_nonzero` is applied to `diff`, so any constant value should be handled.

**Comment by tupui** [comment]
> `count_nonzero` is applied to `diff`, so any constant value should be handled.

Yes, but then you don't want to return 0 for all.

**Comment by k-styles** [comment]
> > `count_nonzero` is applied to `diff`, so any constant value should be handled.
> 
> Yes, but then you don't want to return 0 for all.

@tupui So the change I have made it seems to be okay with you right? On confirmation on this from you, I'll start working on the tests as well.

**Comment by k-styles** [comment]
> Sorry, I don't really know as it's not my main area (and I am not sure who to ping for scipy.integrate. @ev-br is it you?). Hence without tests it's hard to say. You can work on tests without having any features implemented (test driven development).

Oh okay, Sure, I'll be on it then.

**Review by tupui — APPROVED** [review]
The tests make sense to me, thank you @k-styles. I would like a second pair of eyes before merging but LGTM.

**Comment by k-styles** [comment]
@tupui 
`Windows Meson tests / Meson Windows tests (pull_request) ` test seems to be failing.
Is it because of the changes made in the code or is it something else?

**Comment by k-styles** [comment]
@tupui Some workflows have been running for quite a long time. Could you please check on them?

**Comment by tupui** [comment]
I closed-reopened to re-trigger the CI

**Review by tupui — APPROVED** [review]
LGTM now, thank you @k-styles. I will let this open until April 1st to give time for others to look at it. It there are no comments until then I will merge.

**Comment by k-styles** [comment]
> LGTM now, thank you @k-styles. I will let this open until April 1st to give time for others to look at it. It there are no comments until then I will merge.

Thanks @tupui !

**Comment by k-styles** [comment]
@tupui Is it safe to merge this change?

**Comment by tupui** [comment]
Yes thanks for the ping I forgot yesterday

**Pull request merged**
